### PR TITLE
Refactor CreateCourseRunView and CreateRunFromDashboardView

### PR DESCRIPTION
**Pull request by amangano-edx**
Refactors the publisher CreateCourseRunView CreateRunFromDashboardView(s) to be more consistent and make it easier to add logic to handle entitlement-aware courses.

Still need to add/verify tests.

### CODE REVIEW

**Review by mikix — APPROVED** [review]
Seems fine to me.  Refactor diffs are a little hard to follow, but the changes make sense.

You grouped the form UI options a bit differently?  Merged start date and end date and left type/mode separate from end date? If so, nice. :+1:

**Comment by amangano-edx** [comment]
@mikix Yes, I move the form inputs around so that the inputs for each Django form-type are together.
